### ISSUE TITLE
No sound output at all in v4.0.0 at Ubuntu (pipewire)

### ISSUE THREAD

**Issue by Sfinx** [telegramdesktop/tdesktop#24662]
### Steps to reproduce

1. Upgrade from 3.x to 4.x at Ubuntu with pipewire audio subsystem
2. Go to settings / sound, try to play any sound
3. No sound will be heard

At telegram logs I see

```
[2022.06.21 16:54:17] Audio Playback Devices: Dummy Output
[2022.06.21 16:54:17] Audio Playback Default Device: Dummy Output
```

But my system do not have such device:

```
pactl list short sinks 
0	alsa_output.pci-0000_09_00.1.hdmi-stereo	module-alsa-card.c	s16le 2ch 44100Hz	SUSPENDED
1	alsa_output.pci-0000_0b_00.3.analog-stereo	module-alsa-card.c	s16le 2ch 44100Hz	SUSPENDED
3	bluez_sink.xx_xx_xx_xx_xx_xx.a2dp_sink	module-bluez5-device.c	s16le 2ch 44100Hz	RUNNING
```

All other application except telegram v4.0.0 having normal sound output and works as expected.

### Expected behaviour

Sound output must work

### Actual behaviour

No sound output in telegram at all, nor messages, nor videos, nor notifications

### Operating system

Ubuntu 20.04 with pipewire audio subsystem

### Version of Telegram Desktop

v4.0.0

### Installation source

Static binary from official website

### Logs

```shell
[2022.06.21 16:54:16] Executable dir: /usr/bin/, name: telegram-desktop
[2022.06.21 16:54:16] Initial working dir: /home/rus/
[2022.06.21 16:54:16] Working dir: /home/rus/.local/share/TelegramDesktop/
[2022.06.21 16:54:16] Command line: telegram-desktop --
[2022.06.21 16:54:16] Executable path before check: /usr/bin/telegram-desktop
[2022.06.21 16:54:16] Logs started
[2022.06.21 16:54:16] Launcher filename: appimagekit_3a48058ef277d45ae2228089429c0259-Telegram_Desktop.desktop
[2022.06.21 16:54:16] Getting DE, XDG_CURRENT_DESKTOP: 'ubuntu:GNOME'
[2022.06.21 16:54:16] Getting DE, XDG_SESSION_DESKTOP: 'ubuntu'
[2022.06.21 16:54:16] Getting DE, DESKTOP_SESSION: 'ubuntu'
[2022.06.21 16:54:16] Getting DE via WM: 'GNOME Shell'
[2022.06.21 16:54:16] Getting DE, GNOME_DESKTOP_SESSION_ID: 'this-is-deprecated'
[2022.06.21 16:54:16] Getting DE, KDE_FULL_SESSION: ''
[2022.06.21 16:54:16] DE: Gnome
[2022.06.21 16:54:16] Connecting local socket to /run/user/1000/76de3e093c7e7e29ad1984900319463b-{87A94AB0-E370-4cde-98D3-ACC110C5967D}...
[2022.06.21 16:54:16] This is the only instance of Telegram, starting server and app...
[2022.06.21 16:54:16] Moved logging from '/home/rus/.local/share/TelegramDesktop/log_start0.txt' to '/home/rus/.local/share/TelegramDesktop/log.txt'!
[2022.06.21 16:54:16] Primary screen DPI: 126
[2022.06.21 16:54:16] System tray available: [TRUE]
[2022.06.21 16:54:16] Font: from ':/gui/fonts/DAOpenSansRegular.ttf' loaded 'DAOpenSansRegular'
[2022.06.21 16:54:16] Font: from ':/gui/fonts/DAVazirRegular.ttf' loaded 'DAVazirRegular'
[2022.06.21 16:54:16] Font: from ':/gui/fonts/DAOpenSansRegularItalic.ttf' loaded 'DAOpenSansRegularItalic'
[2022.06.21 16:54:16] Font: from ':/gui/fonts/DAVazirRegular.ttf' loaded 'DAVazirRegular'
[2022.06.21 16:54:16] Font: from ':/gui/fonts/DAOpenSansSemiboldAsBold.ttf' loaded 'DAOpenSansSemibold'
[2022.06.21 16:54:16] Font: from ':/gui/fonts/DAVazirMediumAsBold.ttf' loaded 'DAVazirMedium'
[2022.06.21 16:54:16] Font: from ':/gui/fonts/DAOpenSansSemiboldItalicAsBold.ttf' loaded 'DAOpenSansSemiboldItalic'
[2022.06.21 16:54:16] Font: from ':/gui/fonts/DAVazirMediumAsBold.ttf' loaded 'DAVazirMedium'
[2022.06.21 16:54:16] Font: from ':/gui/fonts/DAOpenSansSemiboldAsBold.ttf' loaded 'DAOpenSansSemibold'
[2022.06.21 16:54:16] Font: from ':/gui/fonts/DAVazirMediumAsBold.ttf' loaded 'DAVazirMedium'
[2022.06.21 16:54:16] Font: from ':/gui/fonts/DAOpenSansSemiboldItalicAsBold.ttf' loaded 'DAOpenSansSemiboldItalic'
[2022.06.21 16:54:16] Font: from ':/gui/fonts/DAVazirMediumAsBold.ttf' loaded 'DAVazirMedium'
[2022.06.21 16:54:16] Icon theme: Adwaita
[2022.06.21 16:54:16] Fallback icon theme: 
[2022.06.21 16:54:16] App Info: reading settings...
[2022.06.21 16:54:16] App Info: reading encrypted settings...
[2022.06.21 16:54:16] Lang Info: Loaded cached, keys: 4026
[2022.06.21 16:54:17] OpenAL Logging Level: (not set)
[2022.06.21 16:54:17] Audio Playback Devices: Dummy Output
[2022.06.21 16:54:17] Audio Playback Default Device: Dummy Output
[2022.06.21 16:54:17] Audio Capture Devices: Webcam C310 Mono;Monitor of Dummy Output
[2022.06.21 16:54:17] Audio Capture Default Device: Webcam C310 Mono
[2022.06.21 16:54:17] Using Unity launcher counter.
[2022.06.21 16:54:17] App Info: reading accounts info...
[2022.06.21 16:54:17] App Info: bad decrypt key, data not decrypted - incorrect password?
[2022.06.21 16:54:17] App Info: could not decrypt pass-protected key from info file, maybe bad password...
[2022.06.21 16:54:17] OpenGL: Force-disabled.
[2022.06.21 16:54:17] OpenGL: [FALSE] (OverlayWidget)
[2022.06.21 16:54:20] App Info: reading accounts info...
[2022.06.21 16:54:20] App Info: reading encrypted info...
[2022.06.21 16:54:20] App Info: reading map...
[2022.06.21 16:54:20] App Info: reading encrypted map...
[2022.06.21 16:54:20] App Info: reading encrypted user settings...
[2022.06.21 16:54:20] App Info: encrypted user settings read.
[2022.06.21 16:54:20] App Info: reading encrypted mtp data...
[2022.06.21 16:54:20] MTP Info: read keys, current: 5, to destroy: 0
[2022.06.21 16:54:20] Map read time: 16
[2022.06.21 16:54:20] App Info: reading encrypted mtp config...
[2022.06.21 16:54:20] App Info: reading map...
[2022.06.21 16:54:20] App Info: reading encrypted map...
[2022.06.21 16:54:20] App Info: reading encrypted user settings...
[2022.06.21 16:54:20] App Info: encrypted user settings read.
[2022.06.21 16:54:20] App Info: reading encrypted mtp data...
[2022.06.21 16:54:20] MTP Info: read keys, current: 5, to destroy: 0
[2022.06.21 16:54:20] Map read time: 0
[2022.06.21 16:54:20] App Info: reading encrypted mtp config...
[2022.06.21 16:54:20] App Info: reading map...
[2022.06.21 16:54:20] App Info: reading encrypted map...
[2022.06.21 16:54:20] App Info: reading encrypted user settings...
[2022.06.21 16:54:20] App Info: encrypted user settings read.
[2022.06.21 16:54:20] App Info: reading encrypted mtp data...
[2022.06.21 16:54:20] MTP Info: read keys, current: 5, to destroy: 0
[2022.06.21 16:54:20] Map read time: 1
[2022.06.21 16:54:20] App Info: reading encrypted mtp config...
[2022.06.21 16:54:20] Export Info: Destroy top bar by controller removal.
[2022.06.21 16:54:21] API Warning: not loaded minimal channel applied.
```

**Comment by Sfinx** [comment]
> What's the version of pipewire? thinking

```
dpkg -l pipewire
Desired=Unknown/Install/Remove/Purge/Hold
| Status=Not/Inst/Conf-files/Unpacked/halF-conf/Half-inst/trig-aWait/Trig-pend
|/ Err?=(none)/Reinst-required (Status,Err: uppercase=bad)
||/ Name           Version                          Architecture Description
+++-==============-================================-============-===================================================
ii  pipewire:amd64 0.3.52.r2.g9255cfb-1~ubuntu20.04 amd64        audio and video processing engine multimedia server
```

**Comment by Sfinx** [comment]
Just switched back to 3.7.3 and sound appeared back. But lost all my settings, f...

**Comment by Sfinx** [comment]
No I can't and do not see any reason to do so. The main issue here is that only telegram and only v4.0.0 do not work. All other applications including chrome, firefox, opera, discord, music & video players, etc. are working ok. I've downgraded the telegram v3.7.3 and this particular version works too. Something was changed exactly in telegram that prevents audio output for this app.

**Comment by ilya-fedin** [comment]
> Something was changed exactly in telegram between v3.7.3 and v4.0.0 that prevents audio output for this app.

openal-soft was updated

**Comment by Sfinx** [comment]
I do not have installed openal-soft in the system at all so your logic about reporting this issue to openal team is totally broken. I can help supplying any additional info if needed but I can't push the bugs fixes for third party software embedded in telegram instead of you. Currently I had to use v3.7.3 as it is has sound output

**Comment by ilya-fedin** [comment]
That's how reporting to open source projects works: you're supposed to find the right project that has the affected code. If you won't report to openal, you will just have the bug until someone other will report it :man_shrugging: 

I'm not going to report as I can't reproduce, sorry.

**Comment by Sfinx** [comment]
And this is how bug fixing works for developers that are responsible for their _own_ decisions and for not responsible ones. Last usually say that "it works for me" and all other is personal user problem. походу не шмогла, хехе

**Comment by dafadey** [comment]
I have
ii  pipewire:amd64                           0.3.51-1
and still no sound in telegram v4.0 on Debian

**Comment by onmeac** [comment]
Also just upgraded to v4 of Telegram desktop, though I am on Almalinux 8 with pipewire-0.3.6-1

Telegram desktop freezes on sound event:

```
$ ./Telegram 
error: : cannot open
error: : cannot open
error: : cannot open
[E][000037766.429833][audioadapter.c:528 impl_node_send_command()] audioadapter 0x7f58bba00728: can't send command: Invalid argument
[E][000037766.429846][impl-node.c:194 start_node()] (Telegram-17) start node error -22: Invalid argument
[W][000037766.429850][remote-node.c:480 client_node_command()] node 0x7f58b609edc0: start failed
Invalid return value 0 for stream protocol
Invalid return value 0 for stream protocol
```

and in /var/log/messages
```
Jun 22 00:51:02 pipewire[3506]: #033[1;33m[W][000037808.563586][private.h:73 ratelimit_test()] 187 events suppressed#033[0m
Jun 22 00:51:02 pipewire[3506]: #033[1;33m[W][000037808.563605][impl-node.c:1399 node_ready()] (Dummy-28) graph not finished: state:0x7f7a3e6c9008 quantum:512 pending 1/3#033[0m
Jun 22 00:51:02 pipewire[3506]: #033[1;33m[W][000037808.563609][impl-node.c:844 dump_states()] (Dummy-28) state:0x7f7a3e6c9008 pending:1/3 s:37808057008704 a:37808066877952 f:37808066878342 waiting:9869248 process:390 status:finished sync:0#033[0m
Jun 22 00:51:02 pipewire[3506]: #033[1;33m[W][000037808.563615][impl-node.c:844 dump_states()] (Dummy-28) state:0x7f7a3e6c9008 pending:1/3 s:37808057008704 a:37808066877952 f:37808066878342 waiting:9869248 process:390 status:finished sync:0#033[0m
Jun 22 00:51:02 pipewire[3506]: #033[1;33m[W][000037808.563619][impl-node.c:844 dump_states()] (Telegram-73) state:0x7f7a3e5b4008 pending:0/1 s:37808066878612 a:0 f:0 waiting:18446706265642673004 process:0 status:triggered sync:0#033[0m

```

related to this bug report or should a create a new one?

**Comment by ilya-fedin** [comment]
You have a bugged pipewire version apparently

**Comment by dafadey** [comment]
Telegram/Telegram 
error: : cannot open
error: : cannot open
error: : cannot open
qt.gui.imageio.jpeg: Corrupt JPEG data: premature end of data segment

ii pipewire:amd64 0.3.51-1
and still no sound in telegram v4.0 on Debian

**Comment by Sfinx** [comment]
> I am using alsa and it looks like the guy who opened the thread also uses alsa, no?

Alsa is used always as the only available audio backend for pulseaudio or pipewire in current Linux. Mine OS is stock Ubuntu. All is working except telegram v4.0. It is just appeared that telegram team thinks that finding and fixing bugs in static linked third party telegram sw is the pure user task - so I do not care about this issue anymore

**Comment by ilya-fedin** [comment]
> Mine OS is stock Ubuntu.

Stock Ubuntu 20.04 has pipewire 0.2, so it's definitely not stock

**Comment by ilya-fedin** [comment]
> I am using alsa and it looks like the guy who opened the thread also uses alsa, no?

Are you sure? Do you have wireplumber or pipewire-media-session installed? Any of them can make pipewire handle audio and openal will use it for audio therefore.

**Comment by dafadey** [comment]
> Are you sure? Do you have wireplumber or pipewire-media-session installed? Any of them can make pipewire handle audio and openal will use it for audio therefore.
Verified that I have pipewire-media-session installed and running, no wireplumber. I do not see pipewire in systemctl and can not kill it.
my kernel is 5.16.0-6-amd64

**Comment by ilya-fedin** [comment]
> I do not see pipewire in systemctl and can not kill it.

Are you sure you have the same issue? Have you checked log.txt?

**Comment by Zauzolkov** [comment]
I faced the same issue after recent update to version 4.0.

Console output:

```
$ ./Telegram
error: : cannot open
error: : cannot open
error: : cannot open
Invalid return value 0 for stream protocol
Invalid return value 0 for stream protocol
```

`journalctl` log:

```
pipewire[3054]: spa.alsa: 'front:0': playback open failed: Device or resource busy
pipewire[3054]: spa.alsa: 'front:0': playback open failed: Device or resource busy
pipewire[3054]: spa.audioadapter: params Spa:Enum:ParamId:EnumFormat: 0:0 (follower format) Device or resource busy
pipewire[3054]: pw.node: (alsa_output.pci-0000_00_1f.3.analog-stereo-45) suspended -> error (Start error: Device or resource busy)
pipewire-media-session[3055]: ms.core: error id:41 seq:2513 res:-16 (Device or resource busy): Start error: Device or resource busy
```

OS: openSUSE Leap 15.3 (GNU/Linux)
Installation source: binary from official website.

Feel free to ask for more details if needed.

**Comment by dafadey** [comment]
> You can use `viewlogs` [cheat code](https://github.com/telegramdesktop/tdesktop/wiki/Cheat-Codes)

I realized that I was never cheating in games...
you know, I found that my sound works now. I think it is because I tried to kill pipewire (I did not test afrewards)

my log now looks as folows:
[2022.06.22 17:27:18] Launched version: 4000000, install beta: [FALSE], alpha: 0, debug mode: [FALSE]
[2022.06.22 17:27:18] Executable dir: /home/dan/Telegram/, name: Telegram
[2022.06.22 17:27:18] Initial working dir: /home/dan/
[2022.06.22 17:27:18] Working dir: /home/dan/.local/share/TelegramDesktop/
[2022.06.22 17:27:18] Command line: /home/dan/Telegram/Telegram
[2022.06.22 17:27:18] Executable path before check: /home/dan/Telegram/Telegram
[2022.06.22 17:27:18] Logs started
[2022.06.22 17:27:18] Launcher filename: appimagekit_XXXXXXXXXXXXXX-Telegram_Desktop.desktop
[2022.06.22 17:27:18] Getting DE, XDG_CURRENT_DESKTOP: ''
[2022.06.22 17:27:18] Getting DE, XDG_SESSION_DESKTOP: 'fluxbox'
[2022.06.22 17:27:18] Getting DE, DESKTOP_SESSION: 'fluxbox'
[2022.06.22 17:27:18] Getting DE via WM: 'Fluxbox'
[2022.06.22 17:27:18] Getting DE, GNOME_DESKTOP_SESSION_ID: ''
[2022.06.22 17:27:18] Getting DE, KDE_FULL_SESSION: ''
[2022.06.22 17:27:18] DE: Other
[2022.06.22 17:27:18] Connecting local socket to /run/user/1000/XXXXXXXXXX-{XXXXXX-XXXX-XXXX-XXXXX-XXXXXXXX}...
[2022.06.22 17:27:18] This is the only instance of Telegram, starting server and app...
[2022.06.22 17:27:18] Moved logging from '/home/dan/.local/share/TelegramDesktop/log_start1.txt' to '/home/dan/.local/share/TelegramDesktop/log.txt'!
[2022.06.22 17:27:18] Old start log 'log_start0.txt' found, deleted: [TRUE]
[2022.06.22 17:27:18] Primary screen DPI: 96
[2022.06.22 17:27:18] System tray available: [TRUE]
[2022.06.22 17:27:18] Font: from ':/gui/fonts/DAOpenSansRegular.ttf' loaded 'DAOpenSansRegular'
[2022.06.22 17:27:18] Font: from ':/gui/fonts/DAVazirRegular.ttf' loaded 'DAVazirRegular'
[2022.06.22 17:27:18] Font: from ':/gui/fonts/DAOpenSansRegularItalic.ttf' loaded 'DAOpenSansRegularItalic'
[2022.06.22 17:27:18] Font: from ':/gui/fonts/DAVazirRegular.ttf' loaded 'DAVazirRegular'
[2022.06.22 17:27:18] Font: from ':/gui/fonts/DAOpenSansSemiboldAsBold.ttf' loaded 'DAOpenSansSemibold'
[2022.06.22 17:27:18] Font: from ':/gui/fonts/DAVazirMediumAsBold.ttf' loaded 'DAVazirMedium'
[2022.06.22 17:27:18] Font: from ':/gui/fonts/DAOpenSansSemiboldItalicAsBold.ttf' loaded 'DAOpenSansSemiboldItalic'
[2022.06.22 17:27:18] Font: from ':/gui/fonts/DAVazirMediumAsBold.ttf' loaded 'DAVazirMedium'
[2022.06.22 17:27:18] Font: from ':/gui/fonts/DAOpenSansSemiboldAsBold.ttf' loaded 'DAOpenSansSemibold'
[2022.06.22 17:27:18] Font: from ':/gui/fonts/DAVazirMediumAsBold.ttf' loaded 'DAVazirMedium'
[2022.06.22 17:27:18] Font: from ':/gui/fonts/DAOpenSansSemiboldItalicAsBold.ttf' loaded 'DAOpenSansSemiboldItalic'
[2022.06.22 17:27:18] Font: from ':/gui/fonts/DAVazirMediumAsBold.ttf' loaded 'DAVazirMedium'
[2022.06.22 17:27:18] Icon theme: Adwaita
[2022.06.22 17:27:18] Fallback icon theme: 
[2022.06.22 17:27:18] App Info: reading settings...
[2022.06.22 17:27:18] App Info: reading encrypted settings...
[2022.06.22 17:27:18] Lang Info: Loaded cached, keys: 4134
[2022.06.22 17:27:18] OpenAL Logging Level: (not set)
[2022.06.22 17:27:18] Audio Playback Devices: ALSA Default;HDA Intel PCH, ALC236 Analog (CARD=PCH,DEV=0);HDA Intel PCH, HDMI 0 (CARD=PCH,DEV=3);HDA Intel PCH, HDMI 1 (CARD=PCH,DEV=7);HDA Intel PCH, HDMI 2 (CARD=PCH,DEV=8);HDA Intel PCH, HDMI 3 (CARD=PCH,DEV=9);HDA Intel PCH, HDMI 4 (CARD=PCH,DEV=10);HDA Intel PCH, HDMI 5 (CARD=PCH,DEV=11);HDA Intel PCH, HDMI 6 (CARD=PCH,DEV=12);HDA Intel PCH, HDMI 7 (CARD=PCH,DEV=13);HDA Intel PCH, HDMI 8 (CARD=PCH,DEV=14);HDA Intel PCH, HDMI 9 (CARD=PCH,DEV=15);HDA Intel PCH, HDMI 10 (CARD=PCH,DEV=16);HDA Intel PCH, HDMI 11 (CARD=PCH,DEV=17)
[2022.06.22 17:27:18] Audio Playback Default Device: ALSA Default
[2022.06.22 17:27:18] Audio Capture Devices: ALSA Default;HDA Intel PCH, ALC236 Analog (CARD=PCH,DEV=0)
[2022.06.22 17:27:18] Audio Capture Default Device: ALSA Default
[2022.06.22 17:27:18] Not using Unity launcher counter.
[2022.06.22 17:27:18] App Info: reading accounts info...
[2022.06.22 17:27:18] App Info: reading encrypted info...
[2022.06.22 17:27:18] App Info: reading map...
[2022.06.22 17:27:18] App Info: reading encrypted map...
[2022.06.22 17:27:18] App Info: reading encrypted user settings...
[2022.06.22 17:27:18] App Info: encrypted user settings read.
[2022.06.22 17:27:18] App Info: reading encrypted mtp data...
[2022.06.22 17:27:18] MTP Info: read keys, current: 5, to destroy: 0
[2022.06.22 17:27:18] Map read time: 1
[2022.06.22 17:27:18] App Info: reading encrypted mtp config...
[2022.06.22 17:27:18] Export Info: Destroy top bar by controller removal.
[2022.06.22 17:27:18] OpenGL Profile: Compatibility.
[2022.06.22 17:27:18] OpenGL Renderer: Mesa Intel(R) Xe Graphics (TGL GT2)
[2022.06.22 17:27:18] OpenGL Vendor: Intel
[2022.06.22 17:27:18] OpenGL Version: 4.6 (Compatibility Profile) Mesa 20.3.5
[2022.06.22 17:27:18] OpenGL Extensions: GL_ARB_clear_buffer_object, GL_ARB_multi_bind, GL_ARB_query_buffer_object, ... <you know... thousands of them>
[2022.06.22 17:27:18] OpenGL: QOpenGLContext created, version: 4.4.
[2022.06.22 17:27:18] OpenGL: [TRUE] (OverlayWidget)
[2022.06.22 17:27:26] API Warning: not loaded minimal channel applied

NOTE: THIS IS A LOG FOR WORKING SOUND! (after pipeire was terminated and auto-restarted)

**Comment by dafadey** [comment]
The only thing I did to make it work is attempted to kill pipewire and pipewire-media-session. I rerun right away with new PIDs.
I attached my log because exactly with that system sound was not working. I hope to have the same issue after restart (doing it very rarely though, but I will report).

**Comment by mmoussikhine** [comment]
Stock 20.04.4. Fully updated. After the update to 4.0, the sound has dissapeared.

**Comment by ilya-fedin** [comment]
There's no pipewire 0.3 in stock Ubuntu, folks. If you have pipewire 0.3, you have modified your Ubuntu setup and connected unofficial PPAs.

**Comment by mmoussikhine** [comment]
Where did I say that I have pipewire 0.3 or any other version? I don't, that is the point.

**Comment by ilya-fedin** [comment]
Well, as you can see from the issue title & content, this issue is about openal is unable to find devices with pipewire 0.3. If you don't have pipewire, it's not your issue.

**Comment by mmoussikhine** [comment]
Well, my issue is that after 4.0 update, I have no sound. Which is the whole point of this thread, pipewire or not. Would it help to create another issue about this?

**Comment by rhelmke** [comment]
So I stumbled over this issue (or some variation of it) some time ago and just managed to fix it for me. Maybe this helps someone with similar issues.

### System Info
*(No Ubuntu, shouldn't really matter here):*

```plain
distribution: Manjaro
kernel: 5.15.49
pipewire: 0.3.52 (arch repos)
wireplumber: 0.4.10 (arch repos)
telegram-desktop: 4.0.2 (binary vendor distribution)
audio interfaces:
1. Primary: Behringer BCD3000 USB Interface, Multiple Outputs, Single Mic Input
2. Notebook Internal: AMD Family 17h/19h HD Audio Controller --> ALC257
```

### (My) Observations
Using my primary usb audio interface, Telegram picks up my mic input just fine and even plays notifications on new messages. But it just simply refuses to play any voice, media, or call audio.

Now what I found was when I switch my system default playback device to the notebook's audio controller, the integrated speakers just work fine in every way after telegram restart.

### (My) Assumption
Keep in mind that my preferred interface has multiple outputs.

The previously mentioned OpenAL API that is integrated in Telegram for audio handling has implemented native pipewire support. But what if it's simply not able to select the correct output channel? I mean, the single mic input channel works as expected..

### Fix Description
When OpenAL auto-selects the (presumably buggy) native pipewire interface, maybe we can instruct it to use another one that works properly.. We could, e.g., use ALSA, which is always present and can be [routed through pipewire too](https://wiki.archlinux.org/title/PipeWire#ALSA_clients)!!

So lets to that.. check out this strace output of telegram-deskop:

```bash
$ strace telegram-desktop
# [...]
openat(AT_FDCWD, "/etc/openal/alsoft.conf", O_RDONLY) = -1 ENOENT (No such file or directory)
openat(AT_FDCWD, "/etc/xdg/alsoft.conf", O_RDONLY) = -1 ENOENT (No such file or directory)
openat(AT_FDCWD, "/home/<user>/.alsoftrc", O_RDONLY) = -1 ENOENT (No such file or directory)
openat(AT_FDCWD, "/home/<user>/.config/alsoft.conf", O_RDONLY) = -1 ENOENT (No such file or directory)
# [...]
```
Cool - Telegram's statically linked OpenAL API wants to read [config files](https://github.com/kcat/openal-soft/blob/master/alsoftrc.sample) on startup.. we can switch the backend using the `driver` directive ([docs](https://github.com/kcat/openal-soft/blob/abff69e27640d8731cbe37f21b365b34d73dd8b5/alsoftrc.sample#L46-L53))

### TL;DR

```bash
echo "[general]\ndriver = alsa" >> ~/.alsoftrc
# restart telegram-desktop
```

Sound may work now. (at least it did for me)

**Comment by mmoussikhine** [comment]
Thank you, you are a star! Small typo, it should be `drivers`, not `driver`:
```
[general]
drivers = alsa
```
